plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -514,17 +512,28 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
                 // Databricks stores column statistics with physical name
                 physicalName = metadata.get("delta.columnMapping.physicalName").asText();
                 verify(!isNullOrEmpty(physicalName), "physicalName is null or empty");
-                physicalColumnType = buildType(typeManager, typeNode, true);
                 break;
             case NAME:
                 physicalName = metadata.get("delta.columnMapping.physicalName").asText();
-                verify(!isNullOrEmpty(physicalName), "physicalName is null or empty");
-                physicalColumnType = buildType(typeManager, typeNode, true);
                 break;
             default:
                 physicalName = fieldName;
-                physicalColumnType = columnType;
         }
+        // struct cannot be a partition column
+        if (metadata.isEmpty() && isStruct(typeNode)) {
+            verifyStructTypeChanges(typeNode);
+        }
```

Comment:
This should be done in `verifyTypeChanges`, or a new method. This code is slightly duplicated inside `verifyStuctTypeChanges`.